File: BlotoutAnalytics/Foundation/BOFLogs/BOFLogs.m
//
//  BOFLogs.m
//  BlotoutAnalytics
//
//  Copyright © 2019 Blotout. All rights reserved.
//

#import "BOFLogs.h"
#import <Foundation/Foundation.h>

static id sBOFLogsSharedInstance = nil;

void BOFLogDebug(NSString *frmt, ...) {
  if (![BOFLogs sharedInstance].isSDKLogEnabled) {
    return;
  }

  @autoreleasepool {
    va_list args;
    va_start(args,frmt);
    
    NSString * msg = [[NSString alloc] initWithFormat:frmt  arguments:args];
    va_end(args);
    NSString *logMessage = [NSString stringWithFormat:@"[File Name : %s] [Method Name: %s] [Line No: %d] %@", __FILE_NAME__, __PRETTY_FUNCTION__ , __LINE__, msg];
    NSLog(@"Info: %@",logMessage);
  }
}

void BOFLogError(NSString *frmt, ...) {
  if (![BOFLogs sharedInstance].isSDKLogEnabled) {
    return;
  }
    
  @autoreleasepool {
    va_list args;
    va_start(args,frmt);
    
    NSString * msg = [[NSString alloc] initWithFormat:frmt  arguments:args];
    va_end(args);
    NSString *logMessage = [NSString stringWithFormat:@"[File Name : %s] [Method Name: %s] [Line No: %d] %@", __FILE_NAME__, __PRETTY_FUNCTION__ , __LINE__, msg];
    NSLog(@"Info: %@",logMessage);
  }
}

void BOFLogInfo(NSString *frmt, ...) {
  if (![BOFLogs sharedInstance].isSDKLogEnabled) {
    return;
  }
  
  @autoreleasepool {
    va_list args;
    va_start(args,frmt);
    
    NSString * msg = [[NSString alloc] initWithFormat:frmt  arguments:args];
    va_end(args);
    NSString *logMessage = [NSString stringWithFormat:@"[Method Name: %s] [Line No: %d] %@",__PRETTY_FUNCTION__ , __LINE__, msg];
    NSLog(@"Info: %@",logMessage);
  }
}

@implementation BOFLogs

+ (nullable instancetype)sharedInstance {
  static dispatch_once_t bofLogsOnceToken = 0;
  dispatch_once(&bofLogsOnceToken, ^{
    sBOFLogsSharedInstance = [[[self class] alloc] init];
  });
  
  return sBOFLogsSharedInstance;
}

@end

File: BlotoutAnalytics/Foundation/BOFLogs/BOFLogs.swift
//
//  BOFLogs.swift
//  BlotoutAnalyticsSDK
//
//  Created by Poonam Tiwari on 23/03/22.
//

import Foundation

private var sBOFLogsSharedInstance: Any? = nil

class BOFLogs:NSObject {
    
    var isSDKLogEnabled = false
    public static let sharedInstance = BOFLogs()
    public override init() {}
}

func BOFLogDebug(frmt: String, args:CVarArg...) {
    
    if !BOFLogs.sharedInstance.isSDKLogEnabled  {
        return
    }
    
    let msg =  String(format: frmt , arguments: args)
    let logMessage = "[File Name : \(#file)] [Method Name: \(#function)] [Line No: \(#line)] \(msg)"
    print("Info: \(logMessage)")
    
}

func BOFLogError(_ frmt: String, args:CVarArg...) {
    if !(BOFLogs.sharedInstance.isSDKLogEnabled ) {
        return
    }
    let msg =  String(format: frmt , arguments: args)
    let logMessage = "[File Name : \(#file)] [Method Name: \(#function)] [Line No: \(#line)] \(msg)"
    print("Info: \(logMessage)")
    
}

func BOFLogInfo(_ frmt: String, args:CVarArg...) {
    if !(BOFLogs.sharedInstance.isSDKLogEnabled ) {
        return
    }
    let msg =  String(format: frmt , arguments: args)
    let logMessage = "[Method Name: \(#function)] [Line No: \(#line)] \(msg)"
    print("Info: \(logMessage)")
    
}

